### PR TITLE
Untested complete_and_broadcast impl

### PR DESCRIPTION
**Pull request by AloeareV**

### CODE REVIEW

File: zingolib/src/lightclient/send.rs [hunk 1]
```diff
@@ -133,16 +133,18 @@ impl LightClient {
 #[cfg(feature = "zip317")]
 /// patterns for newfangled propose flow
 pub mod send_with_proposal {
-    use std::convert::Infallible;
+    use std::{convert::Infallible, ops::DerefMut as _};
```
Comment (fluidvanadium) [inline zingolib/src/lightclient/send.rs]: can we just use DerufMut without the _?

Comment (AloeareV) [reply zingolib/src/lightclient/send.rs]: We can, I like the convention of using `_` for traits that aren't named in the file, to indicate that they are imported for side effects (i.e. getting their trait methods into scope).

File: zingolib/src/lightclient/send.rs [hunk 2]
```diff
@@ -199,6 +220,61 @@ pub mod send_with_proposal {
...
+            let (sapling_output, sapling_spend): (Vec<u8>, Vec<u8>) = self
+                .read_sapling_params()
+                .map_err(CompleteAndBroadcastError::SaplingParams)?;
+            let sapling_prover = LocalTxProver::from_bytes(&sapling_spend, &sapling_output);
+            let unified_spend_key =
+                UnifiedSpendingKey::try_from(self.wallet.wallet_capability().as_ref())
+                    .map_err(CompleteAndBroadcastError::UnifiedSpendKey)?;
+
+            let mut step_results = Vec::with_capacity(proposal.steps().len());
+            for step in proposal.steps() {
```
Comment (zancas) [inline zingolib/src/lightclient/send.rs]: This is premature support for 320.   Let's remove it.

File: zingolib/src/lightclient/send.rs [hunk 3]
```diff
@@ -199,6 +220,61 @@ pub mod send_with_proposal {
...
+                        step,
+                    )
+                    .map_err(CompleteAndBroadcastError::Calculation)?
+                };
+
+                step_results.push((step, step_result));
+            }
+            for (_step, step_result) in step_results {
+                self.wallet
+                    .send_to_addresses_inner(
```
Comment (fluidvanadium) [inline zingolib/src/lightclient/send.rs]: we need to digest sent transaction BEFORE calculating the second proposal
AND ideally we ougta digest BEFORE broadcasting

**Review by fluidvanadium — CHANGES_REQUESTED** [review]

File: zingolib/src/lightclient/send.rs (hunk 3 above)
Comment (fluidvanadium) [reply zingolib/src/lightclient/send.rs]: this is stable or upgrade

**Review by fluidvanadium — DISMISSED** [review]
excellente!! incredible!

**Comment by fluidvanadium** [comment]
oh no,  failed due to a merge conflict

File: zingolib/src/lightclient/send.rs [hunk 4]
```diff
@@ -137,25 +137,49 @@ impl LightClient {
 #[cfg(feature = "zip317")]
 /// patterns for newfangled propose flow
 pub mod send_with_proposal {
```
Comment (Oscar-Pepper) [inline zingolib/src/lightclient/send.rs]: i cant see this mod in the changelog, if its added all other pub in this mod will be covered implicitly

**Review by Oscar-Pepper — DISMISSED** [review]

File: zingolib/src/lightclient/send.rs (hunk 4 above)
Comment (AloeareV) [reply zingolib/src/lightclient/send.rs]: Perhaps....but this PR adds no public methods to the interface.

**Comment by fluidvanadium** [comment]
is `fulfill` a better word than 'complete' for changing a Proposal into a Transaction?

File: zingolib/src/lightclient/send.rs (hunk 4 above)
Comment (Oscar-Pepper) [reply zingolib/src/lightclient/send.rs]: well my point is it changes `pub enum CompleteAndBroadcastError` and therefore the quest of how do i update the changelog should resolve at "oops this whole mod was not added in the first place"

Comment (AloeareV) [reply zingolib/src/lightclient/send.rs]: Oh right, the error is public. D'oh.

**Review by zancas — APPROVED** [review]

**Pull request merged**